File: backend/app/schemas/settlement.py
from decimal import Decimal
from pydantic import BaseModel

class TransferSuggestion(BaseModel):
    from_user_id: int
    to_user_id: int
    amount: Decimal

class SettlementReport(BaseModel):
    shamsi_year: int
    shamsi_month: int
    transfers: list[TransferSuggestion]

File: frontend/types/settlement.ts
export type Transfer = {
  from_user_id: number;
  to_user_id: number;
  amount: string;
};

export type Balance = {
  user_id: number;
  balance: string;
};

export type SettlementReport = {
  balances: Balance[];
  my_balances: Balance[];
  transfers: Transfer[];
};
